runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/interceptors/InterceptorExecutor.kt

```diff
@@ -328,4 +336,22 @@ internal class InterceptorExecutor<I, O>(
             },
         )
     }
+
+    private fun updateBusinessMetrics(
```

Comment:
suggestion: Instead of calling this function after almost every interceptor stage, can it just be added as a new interceptor? It can keep an internal `var request : HttpRequest` that is updated at each stage, and then you can use the `readBeforeX` interceptor hooks to compare the `modifiedRequest`

Reply:
Any thoughts on this suggestion? I think it would be cleaner, `InterceptorExecutor` shouldn't really be responsible for emitting / updating business metrics IMO

Reply:
Currently we're only checking if the URL changed after the `modifyX` interceptors are executed. We could instead add a new interceptor right after in the `readY` phase to check if the URL had changed.

It's not a one-way decision so I'm fine to leave it as it is and refactor later if needed.